### DISCUSSION TITLE
Upgrading and pip version

### DISCUSSION THREAD

**Discussion by Kludge-As-A-Service** [netbox-community/netbox#7489] [Q&A] [answered]
Hello, I plan to upgrade from 2.9.11 to 2.11.12 to 3.0.6. As we are using Redhat 7 which only provides python 3.6, I made a manual install of python 3.9 to satisfy netbox requirements. I have noticed in the upgrade doc that the path of the appropriate python binary can be provided to the upgrade script so everything is fine regarding that, but what about pip? As of now, regular pip command runs pip2.7 it seems, and we have also pip3.6 and pip3.9 at our disposal, but which one will netbox pick, do I have to force the use of pip3.9 somewhere, or does it even matter? 
Could someone please clarify this point? Thanks.

**Comment by cpmills1975** [comment, marked as answer]
pip itself runs within the virtualenv so it's possible to have pip3 alongside your python3 installation. The upgrade script launches the virtualenv before it does anything else.
